youtube_dl/extractor/bandcamp.py

```diff
@@ -48,6 +48,10 @@ def _real_extract(self, url):
             if m_trackinfo:
                 json_code = m_trackinfo.group(1)
                 data = json.loads(json_code)[0]
+                track_id = compat_str(data['id'])
+
+                if not data['streaming']:
+                    raise ExtractorError('Not streamable', video_id=track_id, expected=True)
 
                 formats = []
                 for format_id, format_url in data['file'].items():
```

Comment:
Instead of relying on another heuristic that may not even always work why not just make this line not to crash when no proper `file` is available?